src/main/java/net/ripe/rpki/commons/ta/serializers/TrustAnchorRequestSerializer.java

```diff
@@ -190,8 +190,14 @@ public TrustAnchorRequest deserialize(final String xml) {
                 throw new DomXmlSerializerException("creationTimestamp content is not a number: " + creationTimeStampText, e);
             }
 
-            final Element taCertificatePublicationUriElement = getSingleChildElement(taRequestElement, TA_CERTIFICATE_PUBLICATION_URI);
-            final URI taCertificatePublicationUri = URI.create(getElementTextContent(taCertificatePublicationUriElement));
+            // The TA_CERTIFICATE_PUBLICATION_URI is present only for TA exchanges, but for some reason this parser is used for exchanges between other CA inside core
+            final Optional<Element> taCertificatePublicationUriElement = getOptionalSingleChildElement(taRequestElement, TA_CERTIFICATE_PUBLICATION_URI);
+            final URI taCertificatePublicationUri;
```

Comment:
```suggestion
              final URI taCertificatePublicationUri = taCertificatePublicationUriElement
                    .map(uri -> URI.create(getElementTextContent(taCertificatePublicationUriElement.get())))
                    .orElse(null);
```

My Idea setup is a bit kaput, but this should do the same if you can test.

Reply:
I've applied your changes. it works.